pkg/controller/secret/secretsync.go

```diff
@@ -0,0 +1,208 @@
+)
+
+// This SecretSyncController translates Secret provided by cloud-credential-operator into
+// format required by the CSI driver.
+type SecretSyncController struct {
+	operatorClient  v1helpers.OperatorClient
+	kubeClient      kubernetes.Interface
+	secretLister    corelisters.SecretLister
+	configMapLister corelisters.ConfigMapLister
+	eventRecorder   events.Recorder
+}
+
+const (
+	// Name of key with ibmcloud_api_key in Secret provided by cloud-credentials-operator.
+	cloudSecretKey = "ibmcloud_api_key"
+	// Name of key with cloud.conf in ConfigMap provided by cloud-credentials-operator.
+	CloudConfigmapKey = "cloud.conf"
+	// Name of the key in secret storage-secret-store creating on operator namespace
+	StorageSecretStoreKey = "slclient.toml"
+
+	// storage-secret-store data format
+	StorageSecretTomlTemplate = `[vpc]
+iam_client_id = "bx"
+iam_client_secret = "bx"
+
+g2_token_exchange_endpoint_url = "https://iam.bluemix.net"
+g2_riaas_endpoint_url = "https://%s.iaas.cloud.ibm.com"
+
+g2_resource_group_id = "%s" 
+g2_api_key = "%s"
+provider_type = "g2"
+`
+)
+
+func NewSecretSyncController(
+	operatorClient v1helpers.OperatorClient,
+	kubeClient kubernetes.Interface,
+	informers v1helpers.KubeInformersForNamespaces,
+	resync time.Duration,
+	eventRecorder events.Recorder) factory.Controller {
+
+	// Read secret from operator namespace and save the translated one to the operand namespace
+	secretInformer := informers.InformersFor(util.OperatorNamespace)
+	configMapInformer := informers.InformersFor(util.ConfigMapNamespace)
+	c := &SecretSyncController{
+		operatorClient:  operatorClient,
+		kubeClient:      kubeClient,
+		secretLister:    secretInformer.Core().V1().Secrets().Lister(),
+		configMapLister: configMapInformer.Core().V1().ConfigMaps().Lister(),
+		eventRecorder:   eventRecorder.WithComponentSuffix("SecretSync"),
+	}
+	return factory.New().WithSync(c.sync).ResyncEvery(resync).WithSyncDegradedOnError(operatorClient).WithInformers(
+		operatorClient.Informer(),
+		secretInformer.Core().V1().Secrets().Informer(),
+		configMapInformer.Core().V1().Secrets().Informer(),
+	).ToController("SecretSync", eventRecorder)
+}
+
+func (c *SecretSyncController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	opSpec, _, _, err := c.operatorClient.GetOperatorState()
+	if err != nil {
+		klog.V(2).Infof("Error while getting operator state %s", err)
+		return err
+	}
+	if opSpec.ManagementState != operatorv1.Managed {
+		klog.V(2).Infof("Operator management state is not managed")
+		return nil
+	}
+
+	cloudSecret, err := c.secretLister.Secrets(util.OperatorNamespace).Get(util.CloudCredentialSecretName)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// TODO: report error after some while?
+			klog.V(2).Infof("Waiting for secret %s from %s", util.CloudCredentialSecretName, util.OperatorNamespace)
+			return nil
+		}
+		klog.V(2).Infof("Secret listener failed to get secret details %s", err)
```

Comment:
In these cases, where an error is being returned, the logs should probably be `Errorf` instead of `Infof` to differentiate real errors from informational logging.